### PR TITLE
[META-86] Move sorting & filtering to graphql queries

### PR DESCRIPTION
**Pull request by kigadye**
### Description of issue
Move sorting & filtering to graphql queries because Supabase returns max. 30 entries per requests.

Still missing pagination but technically works already because if you split up trending projects into daily/ weekly/ monthly every query actually has < 30 results. Will add pagination in a separate PR.

### How I implemented
- Used existing UI from previous implementation
- Adjusted everything to work with GraphQL queries
- Improved/ unified dropdown styling

### Other
Refactored code:
- Split up a lot of large files into smaller ones
- Moved projects table to side-effects folder
- Deleted unused files
- Still a lot to do/ clean up

### CODE REVIEW

**Review by ArnoClaude — APPROVED** [review]
Wow, great improvements, well done!

File: src/components/page/overview/FilterItemModal.tsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+type FilterItemModalProps = {
+  currentKey: FilterOption['key']
+  filters: ProjectFilter
+  updateFilters: (filter: ProjectFilter) => void
+}
+
+const FilterItemModal = ({ filters, currentKey, updateFilters }: FilterItemModalProps) => {
+  const type = filterOptions.find((option) => option.key === currentKey)?.type
+  const column = filterOptions.find((option) => option.key === currentKey)?.column
+  const currentOperator = Object.keys(filters[currentKey] || {})[0] as
```
Comment (pczern) [inline src/components/page/overview/FilterItemModal.tsx:22]: this smells

File: src/components/page/overview/EditColumns.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+const EditColumns: FC<EditColumnsProps> = ({ columns }) => {
+  const [open, setOpen] = useState<boolean>(false)
+
+  return (
+    <Menu as="div" className="relative">
+      <Menu.Button as="div">
+        <Button
+          Icon={TbColumns2}
```
Comment (pczern) [inline src/components/page/overview/EditColumns.tsx:21]: 2?

File: src/components/page/overview/EditColumns.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import { Project } from '@/graphql/generated/gql'
+import Button from '@/components/pure/Button'
+import MenuItemsTransition from './MenuItemsTransition'
+
+type EditColumnsProps = {
+  columns: Column<Project, unknown>[]
+}
+
+const EditColumns: FC<EditColumnsProps> = ({ columns }) => {
+  const [open, setOpen] = useState<boolean>(false)
```
Comment (pczern) [inline src/components/page/overview/EditColumns.tsx:15]: is this not implicitly typed

File: src/components/page/overview/FilterBar.tsx [hunk 4]
```diff
@@ -1,41 +1,49 @@
...
+        <div className="my-auto h-4/5 border-l border-gray-800" />
+      )}
+
+      {/* Filters (exclude time filter options) */}
+      {Object.keys(filters)
+        .filter((key) => !Object.values(TimeFilterOption).includes(key as TimeFilterOption))
+        .map((key) => (
+          <FilterItemModal
+            key={key}
+            currentKey={key as FilterOption['key']}
```
Comment (pczern) [inline src/components/page/overview/FilterBar.tsx:39]: code smell

File: src/components/page/overview/FilterItemModal.tsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    const newFilters = { ...filters }
+    newFilters[currentKey] = {
+      [operator]: type === 'string' ? value : parseInt(value as string, 10)
+    }
+    updateFilters(newFilters)
+  }
+
+  const removeFilter = () => {
+    const newFilters = { ...filters }
+    delete newFilters[currentKey]
```
Comment (pczern) [inline src/components/page/overview/FilterItemModal.tsx:45]: you can probably just use null in the object before

File: src/components/page/overview/FilterItemModal.tsx [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    currentOperator
+  )
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => setValue(e.target.value)
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+
+    const newFilters = { ...filters }
+    newFilters[currentKey] = {
```
Comment (pczern) [inline src/components/page/overview/FilterItemModal.tsx:37]: you can do this in the object in line 36

File: src/components/page/overview/TimeFrameSelector.tsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+      <Button Icon={AiOutlineCalendar} variant="normal" text={currentTimeFilterLabel} order="ltr" />
+    </Menu.Button>
+
+    <MenuItemsTransition>
+      <Menu.Items className="absolute left-0 z-30 mt-2 origin-top-right rounded-[5px] bg-gray-700 p-1 shadow-lg focus:outline-none">
+        {Object.entries(TimeFilterOption).map(([label, timeFrame]) => (
+          <Menu.Item
+            as="button"
+            key={timeFrame}
+            onClick={() => handleChange(timeFrame)}
```
Comment (pczern) [inline src/components/page/overview/TimeFrameSelector.tsx:31]: handle should not be arrowed

Comment (pczern) [reply src/components/page/overview/TimeFrameSelector.tsx:31]: that is the usual convention imho

File: src/components/page/overview/TopBar.tsx [hunk 8]
```diff
@@ -1,275 +1,91 @@
...
-                      {column.getIsVisible() ? (
-                        <RiCheckboxFill className="text-indigo-600" />
-                      ) : (
-                        <RiCheckboxBlankLine />
-                      )}
+    // Remove all other time filters
+    const newFilters = { ...filters }
+    Object.values(TimeFilterOption)
+      .filter((t) => t !== timeFrame)
+      .forEach((t) => delete newFilters[t])
```
Comment (pczern) [inline src/components/page/overview/TopBar.tsx:53]: delete is a code smell, as well as forEach

File: src/components/page/overview/TopBar.tsx [hunk 9]
```diff
@@ -1,275 +1,91 @@
...
-              }`}
-            >
-              <TbArrowsSort className="text-gray-500" />
-              <p className={`leading-none ${tableSort ? '' : 'text-gray-500'}`}>Sorting</p>
-            </Menu.Button>
-          </div>
+  // Maps time frame enum value to the corresponding key
+  const currentTimeFilterLabel = Object.entries(TimeFilterOption).filter(
+    (option) => option[1] === selectedTimeFrame
+  )[0][0]
```
Comment (pczern) [inline src/components/page/overview/TopBar.tsx:61]: [0][0] can this be, should this be, is this good design

File: src/components/side-effects/Compare/index.tsx [hunk 10]
```diff
@@ -137,57 +136,60 @@ const Compare = () => {
...
-            Icon={FaSlack}
-            order="ltr"
-            textColor="white"
+      {!fetching && !error && data.length > 0 && (
+        <>
+          <Chart
+            datasets={data.map((project) => ({
+              id: project.id,
+              name: project.name,
+              data: project.starHistory as React.ComponentProps<typeof Chart>['datasets'][0]['data']
```
Comment (pczern) [inline src/components/side-effects/Compare/index.tsx:145]: smells

File: src/components/side-effects/Compare/index.tsx [hunk 11]
```diff
@@ -137,57 +136,60 @@ const Compare = () => {
...
-        filters={filters}
-        setFilteredRowCount={setFilteredRowCount}
-        tableSort={tableSort}
-      />
+      {fetching && <Loading message="Getting trending projects for you..." />}
+
+      {error && <Error />}
+
+      {data.length === 0 && !error && !fetching && (
+        <p className="w-full p-12 text-center text-14 text-gray-300">No projects found</p>
```
Comment (pczern) [inline src/components/side-effects/Compare/index.tsx:189]: an error that is not Error component

**Review by pczern — APPROVED** [review]

File: src/components/side-effects/Compare/index.tsx (hunk 11 above)
Comment (kigadye) [reply src/components/side-effects/Compare/index.tsx:189]: not an error just a placeholder

**Pull request merged**